src/containers/registration/registration.jsx

```diff
@@ -0,0 +1,162 @@
+// @flow
+import React, { Component, Fragment } from "react";
+import styled from "styled-components";
+import Joi from "joi-browser";
+import Input from "../../components/input";
+import Button from "../../components/button";
+import Form from "../../components/form";
+import Alert from "../../components/alert";
+import { type RegistryProfile } from "../../types";
+
+const FormWrapper = styled.section`
+  flex: auto;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  form {
+    max-width: 300px;
+  }
+`;
+type UserErrors = {
+  loginName: string,
+  name: string,
+  surname: string,
+  password: string
+};
+type State = {
+  userData: RegistryProfile,
+  errors: UserErrors
+};
+type Props = {
+  result: string,
+  error: string,
+  loading: boolean,
+  createProfile: (profile: RegistryProfile) => void
+};
+class Registration extends Component<Props, State> {
+  state = {
+    userData: {
+      loginName: "",
+      name: "",
+      surname: "",
+      password: ""
+    },
+    errors: {
+      loginName: "",
+      name: "",
+      surname: "",
+      password: ""
+    }
+  };
+
+  schema = {
+    loginName: Joi.string()
+      .required()
+      .min(6)
+      .label("LoginName"),
+    name: Joi.string()
+      .required()
+      .label("Name"),
+    surname: Joi.string()
+      .required()
+      .label("Surname"),
+    password: Joi.string()
+      .required()
+      .min(8)
+      .label("Password")
+  };
+
+  validate = (): UserErrors => {
+    const errors = {
+      loginName: "",
+      name: "",
+      surname: "",
+      password: ""
+    };
+    const { error } = Joi.validate(this.state.userData, this.schema, {
+      abortEarly: false
+    });
+    if (!error) {
+      return errors;
+    }
+    error.details.forEach(item => {
+      errors[item.path[0]] = item.message;
+    });
+    return errors;
+  };
+
+  handleSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const errors = this.validate();
+    this.setState({ errors });
+    const notErrors = {
+      loginName: "",
+      name: "",
+      surname: "",
+      password: ""
+    };
+    if (JSON.stringify(errors) === JSON.stringify(notErrors)) {
```

Comment:
it is bed solution, think about the following:
https://stackoverflow.com/questions/23604734/how-to-check-if-all-object-keys-has-false-values